src/explore-education-statistics-frontend/src/modules/table-tool/components/TableToolShare.tsx

```diff
@@ -0,0 +1,114 @@
+import ButtonText from '@common/components/ButtonText';
+import Button from '@common/components/Button';
+import ButtonGroup from '@common/components/ButtonGroup';
+import LoadingSpinner from '@common/components/LoadingSpinner';
+import UrlContainer from '@common/components/UrlContainer';
+import { TableHeadersConfig } from '@common/modules/table-tool/types/tableHeaders';
+import mapUnmappedTableHeaders from '@common/modules/table-tool/utils/mapUnmappedTableHeaders';
+import permalinkService from '@common/services/permalinkService';
+import {
+  SelectedPublication,
+  TableDataQuery,
+} from '@common/services/tableBuilderService';
+import ButtonLink from '@frontend/components/ButtonLink';
+import React, { useEffect, useState } from 'react';
+
+interface Props {
+  tableHeaders?: TableHeadersConfig;
+  query: TableDataQuery;
+  selectedPublication: SelectedPublication;
+}
+
+const TableToolShare = ({
+  tableHeaders,
+  query,
+  selectedPublication,
+}: Props) => {
+  const [permalinkId, setPermalinkId] = useState<string>('');
+  const [permalinkLoading, setPermalinkLoading] = useState<boolean>(false);
+
+  useEffect(() => {
+    setPermalinkId('');
+  }, [tableHeaders]);
+
+  const handlePermalinkClick = async () => {
+    if (!tableHeaders) {
+      return;
+    }
+    setPermalinkLoading(true);
+
+    const { id } = await permalinkService.createPermalink(
+      {
+        query,
+        configuration: {
+          tableHeaders: mapUnmappedTableHeaders(tableHeaders),
+        },
+      },
+      selectedPublication.selectedRelease.id,
+    );
+
+    setPermalinkId(id);
+    setPermalinkLoading(false);
+  };
+
+  const handleCopyClick = () => {
+    const el = document.querySelector(
+      "[data-testid='permalink-generated-url']",
+    ) as HTMLInputElement;
+    el?.select();
+    document.execCommand('copy');
+  };
+
+  if (!permalinkId) {
+    return (
+      <>
+        <h3 className="govuk-heading-s">Save table</h3>
+        <LoadingSpinner
+          alert
+          inline
+          loading={permalinkLoading}
+          size="sm"
+          text="Generating permanent link"
+        >
+          <ButtonText onClick={handlePermalinkClick}>
+            Generate shareable link
+          </ButtonText>
+        </LoadingSpinner>
+      </>
+    );
+  }
+
+  return (
+    <div className="dfe-align--left">
```

Comment:
Do we need an `aria-live` here to announce the shareable link has rendered? I'm not entirely sure how the browser responds if a newly rendered element has `aria-live` however.

Reply:
Looks like aria-live needs to be in the DOM from the start, so wouldn't be useful here as it's only rendered as part of the final step. We should consider if/when/how to use it for the table tool in general though.

Reply:
Yeh we should, definitely need to work on the accessibility aspect of it more!